### ISSUE TITLE
`NotificationsPanel` a11y violations

### ISSUE THREAD

**Issue by allisonishida** [carbon-design-system/ibm-products#4940]
## What will this achieve?
Remediate accessibility violations on `NotificationsPanel` component

## How will success be measured?
Pass **IBM Equal Access Accessibility Checker** scan

## Additional information
1. The scan shows `Accessible name does not match or contain the visible label text` for each notification in the panel
![image (74)](https://github.com/carbon-design-system/ibm-products/assets/22247062/17241872-ddee-4a3b-8421-41c75dfe6ecc)
2. The scan shows `Element with "img" role has no label or an empty label` for the empty state image
![image (75)](https://github.com/carbon-design-system/ibm-products/assets/22247062/6807169b-19cb-4145-9201-dfe080fab18a)
